examples/requirements-and-tests/README.md

```diff
@@ -0,0 +1,71 @@
+* How to setup these languages with their services and grammars (file layout).
+* How to create cross language validations (are requirements covered by some tests?).
+* How to create a cross language generator (coverage matrix).
+
+## Language Design
+
+Grammars:
+
+* `common.langium`: common aspects used the main grammars.
+* `requirements.langium`: requirements language (includes `common.langium`)
+* `tests.langium`: requirements language (includes `common.langium` and `requirements.langium`, references `Requirement` elements from `requirements.langium`).
+
+Modules and Services:
+
+* `createRequirementsAndTestsLanguageServices` creates the full set of services used by the CLI and the language server.
+* `requirements-language-module.ts`: requirements service configuration
+* `tests-language-module.ts`: tests service configuration
+
+Language configuration:
+
+* `langium-config.json`: TODO, what does this exactly (AST code generation, syntax highlighting, TextMate).
+* `package.json`: `contributes` section, TODO, what does this exactly (vscode extension linkage?).
+
+CLI and tests:
+
+* `extractDocuments` in `cli-utils.ts` is responsible to load a requirement model file (`*.req`) and all test model files (`*.tst`) located in the folder of the processed requirement file. This is required in order to process the requirement model which needs all associated test models in the workspace.
+
+## Generator CLI
+
+The Example features a generator that you can run via cli to process Requirement Model files. The outcome is a HTML table with a coverage matrix indicating which Test Cases test the Requirements of the file passed to the CLI.
+
+* Ensure the complete project was properly built, otherwise run `npm install` from the root of the Langium project.
+* Use `node ./bin/cli` from the requirements-and-tests directory to run the cli. Follow the instructions or use `node ./bin/cli generate-requirements-coverage <full-path-to-req-file>`.
+
+The generator produces an HTML file.
+
+You also can use `requirements-and-tests-language-cli` as a replacement for `node ./bin/cli`, if you install the cli globally.
+
+* Run `npm install -g ./` from the requirements-and-tests directory.
+* Use `requirements-and-tests-language-cli` to run the cli. Follow the instructions or use `requirements-and-tests-language-cli generate-requirements-coverage <full-path-to-req-file>`.
+
+## VSCode Extension
+
+Please use the VSCode run configuration "Run Requirements and Tests Extension" to launch a new VSCode instance including the extension for this language.
+Use the run configuration "Attach" to attach the debugger.
+
+## Validators
+
+* A test identifier shall contain a number.
+* A requirement identifier shall contain a number.
+* A requirement shall be covered by at least one test (TODO: when updating the test model, the requirement is not rechecked automatically).
```

Comment:
The third point is hard to validate locally because you'd need to scan the whole workspace to find tests that cover a requirement. Typically this would rather be done as additional validation in a CLI or other processing tool, where you know the whole context.

Other idea for validation: a test must specify only environments that are mentioned in the corresponding requirement.